core/src/main/java/org/geysermc/geyser/util/EntityUtils.java

```diff
@@ -209,6 +210,18 @@ public static void updateMountOffset(Entity passenger, Entity mount, boolean rid
                         zOffset = displayTranslation.getZ();
                     }
                 }
+                case PLAYER -> {
+                    if (passenger instanceof  TextDisplayEntity textDisplay) {
+                        Vector3f displayTranslation = textDisplay.getTranslation();
+                        if (displayTranslation != null && textDisplay.getText() != null) {
+                            int lines = PlainTextComponentSerializer.plainText().serialize(textDisplay.getText()).split("\n").length;
```

Comment:
Can this line count not be stored in the text display instead of the whole component?